query-gql/src/lib.rs

```diff
@@ -528,12 +621,17 @@ fn build_parts_for_statements(
                     if let Some(alias) = alias_opt {
                         projection.push(UnaryExpression::alias(expr.clone(), alias.clone()));
                         new_scope.push(UnaryExpression::ident(alias.as_ref()));
-                    } else {
-                        projection.push(expr.clone());
-                        if let Expression::UnaryExpression(UnaryExpression::Identifier(id)) = &expr
-                        {
-                            new_scope.push(UnaryExpression::ident(id));
+                    } else if let Expression::UnaryExpression(UnaryExpression::Identifier(id)) =
+                        &expr
+                    {
+                        if !scope_contains_identifier(&scope, id) {
+                            return Err(QueryParseError::IdentifierNotInScope);
```

Comment:
Should the error include details of the identifier that is not in scope, so that the user has more actionable feedback?